lib/useKWindowDimensions.js

```diff
@@ -0,0 +1,72 @@
+import './composition-api'; //Due to @vue/composition-api shortcomings, add plugin prior to use in kolibri, studio and tests
+import { onBeforeUnmount, onMounted, ref } from '@vue/composition-api';
+
+/** Window dimensions */
+const windowWidth = ref(null);
+const windowHeight = ref(null);
+
+/**
+ * Update window width and height
+ */
+const updateWindow = () => {
+  const metrics = windowMetrics();
+  windowWidth.value = metrics.width;
+  windowHeight.value = metrics.height;
+};
+
+/**
+ * Returns window inner width and height
+ * @returns {Object} The window's inner width and height
+ */
+function windowMetrics() {
+  return {
+    width: window.innerWidth,
+    height: window.innerHeight,
+  };
+}
+
+/**
+ * Check if Nuxt is server side rendering
+ * @returns {Boolean}
+ */
+function isNuxtServerSideRendering() {
+  return process && process.server;
+}
+
+/**
+ * @param {CallableFunction} eventHandler - The event callback function
+ */
+function addWindowListener(eventHandler) {
+  //Only add listeners when not server side rendering
+  if (!isNuxtServerSideRendering()) {
+    if (window.addEventListener) {
+      window.addEventListener('resize', eventHandler, true);
+    } else if (window.attachEvent) {
+      window.attachEvent('onresize', eventHandler);
+    }
+    //First time initialization of dimensions
+    eventHandler();
+  }
+}
+
+/**
+ * @param {CallableFunction} eventHandler - The event callback function
+ */
+function removeWindowListener(eventHandler) {
+  window.removeEventListener(eventHandler);
+}
+
+export default function useKWindowDimensions() {
+  onMounted(() => {
+    addWindowListener(updateWindow);
+  });
+
+  onBeforeUnmount(() => {
+    removeWindowListener(updateWindow);
+  });
```

Comment:
Since `windowWidth` and `windowHeight` are global variables, it seems that this really only needs to `addWindowListener` once and likewise only `removeWindowListener` if there aren't any more active usages of `useKWindowDimensions`. Invoking `removeWindowListener` on `onBeforeUnmount` will likely stop listening all together if this composable is used more than once, meaning one instance will stop working if another is unmounted.

Reply:
Code has been modified to add a listener only if it doesn't exist. Also, A the only added listener can only be removed only if its not in active use else where.